### ISSUE TITLE
`nth-child` not working with `cascade: false`

### ISSUE THREAD

**Issue by cheeaun** [thysultan/stylis.js#28]
I found this bug via using `styled-jsx` and pinpointed the error.

I tried running this test:

```js
'nth-child with cascade false': {
	options: {
		cascade: false
	},
	sample: `b > a:nth-child(2n) { color: blue }`,
	expected: `b.user>a:nth-child(2n){color: blue;}`
}
```

It fails ❌  and the result is this:

```
46 'failed: ' 'test' '\n\n' 'b.user>a.user:nth-child(2n.user){color: blue;}'
36 'expected: ' '\n\n' 'b.user>a:nth-child(2n){color: blue;}' '\n\n---------------\n\n'
```

Somehow the `2n` value got namespaced and became `2n.user` 😢

**Comment by thysultan** [comment]
I see where the problem is, since `:not(selectors)` is namespace that leaks into other selector functions. Will push a patch.

**Comment by thysultan** [comment]
patched in v3.0.10, cc @giuseppeg
